addons/website_blog/controllers/main.py

```diff
@@ -167,6 +171,80 @@ def tags_list(tag_ids, current_tag):
 
+    @http.route(['/blog/<model("blog.blog"):blog>/<any("atom","rss"):feed_type>'], type='http', auth="public", website=True)
+    def blog_feed(self, blog=None, feed_type=None):
+        """ RSS/ATOM
+
+        Blog Detail : fg = FeedGenerator()
+
+        - Id : Blog id (URL)
+        - Title : Blog name
+        - Link : Blog LINK (URL)
+        - Description : Blog Description
+        - Lastbuilddate : Update date (last modification date in the blog) "This will set both ATOM:updated and RSS:lastBuildDate."
+        - Language : blog language
+            ------
+            Blog Post Detail : fe = fg.add_entry()
+            ------
+            - Id : Blog Post id (URL)
+            - Title : Blog Post Name
+            - Link : Blog Post Link (URL) OR Rss Link OR Atom Link
+            - Author : Blog Post Name(Name, Email)
+            - Description : Blog Post Subtitle
+            - Update : last modification date in the Blog Post for use ATOM
+            - Published : Create Date of Blog Post for use ATOM
+            - Pubdate : Create Date of Blog Post for use RSS
+            - Content : Blog Post Content (Diasplay Content In ATOM) "The content MUST be suitable for presentation as HTML"
+        """
+        cr, uid, context = request.cr, request.uid, request.context
+        user = request.registry['res.users'].browse(cr, uid, uid, context=context)
+        base_url = request.registry['ir.config_parameter'].get_param(cr, uid, 'web.base.url')
+        blog_post_obj = request.registry['blog.post']
+
+        blog_url = '%s/blog/%s' % (base_url, slug(blog))
+        fg = FeedGenerator()
+        fg.id(blog_url)
+        fg.title(blog.name)
+        fg.link(href='%s/blog/%s/atom' % (base_url, slug(blog)), type='application/atom+xml', rel='self')
+        fg.link(href='%s/blog/%s/rss' % (base_url, slug(blog)), type='application/rss+xml', rel='self')
+        fg.link(href=blog_url, rel='alternate')
+        fg.description(blog.name)
+        fg.language(user.lang)
+
+        timezone = pytz.timezone(context.get('tz') or 'UTC')
+        blog_write_date = datetime.datetime.strptime(blog.write_date, tools.DEFAULT_SERVER_DATETIME_FORMAT).replace(tzinfo=pytz.utc).astimezone(timezone)
+
+        locale.setlocale(locale.LC_TIME, "en_US.utf8")
+        # This will set both atom:updated and rss:lastBuildDate.
+        fg.lastBuildDate(blog_write_date)
+
+        # Latest updated blogposts
+        blog_post_ids = blog_post_obj.search(cr, uid, [('blog_id', '=', blog.id)], order="write_date desc", limit=15, context=context)
+        for post in blog_post_obj.browse(cr, SUPERUSER_ID, blog_post_ids, context=context):
+            fe = fg.add_entry()
+            blog_post_url = '%s/blog/%s/post/%s' % (base_url, slug(blog), slug(post))
+
+            fe.id(blog_post_url),
+            fe.title(post.name),
+            fe.link(href=blog_post_url),
+            fe.author({'name': post.author_id.name, 'email': post.author_id.email}),
```

Comment:
I'm not sure we always want to expose user's email to spammers... I would remove it